### PR TITLE
protect access to HMICapabilities object with DataAccessor

### LINKED ISSUE #1957: Need to protect the HMICapabilities object from concurrent access
### Bug Report
The global HMICapabilities object was being accessed concurrently by multiple threads, thus causing a SIGSEGV fault.

### PR DESCRIPTION
**Pull request by fronneburg**
Fixes #1957

This PR is **[ready]** for review.

### Risk
This PR makes **[minor]** API changes: instead of returning an HMICapabilities object, the "hmi_capabilities()" API now returns a locked DataAccessor protecting access to the HMICapabilities object. There are also now two versions of this API: hmi_capabilities() and const_hmi_capabilities().

### Testing Plan
All unit tests have been updated and pass

### Summary
All access to the HMICapabilities object is now through a locked DataAccessor.

### CLA
- [X ] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by LuxoftAKutsan** [comment]
@fronneburg please fix build issues

File: src/components/application_manager/test/commands/hmi/ui_get_capabilities_response_test.cc [hunk 1]
```diff
@@ -186,7 +187,7 @@ TEST_F(UIGetCapabilitiesResponseTest, SetNavigation_SUCCESS) {
       CreateCommand<UIGetCapabilitiesResponse>(command_msg));
 
   EXPECT_CALL(app_mngr_, hmi_capabilities())
-      .WillOnce(ReturnRef(mock_hmi_capabilities_));
+      .WillOnce(Return(NonConstDataAccessor<application_manager::HMICapabilities>(mock_hmi_capabilities_, hmi_lock_)));
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/hmi/ui_get_capabilities_response_test.cc]: @fronneburg I would create typedef for NonConstDataAccessor<application_manager::HMICapabilities>

**Comment by LuxoftAKutsan** [comment]
@fronneburg there is still build issues, Please look throw http://opensdl-jenkins.luxoft.com:8080/job/OpenSDL_Develop_PR/job/OpenSDL_PR/1108/

**Comment by fronneburg** [comment]
Sorry about all of this. We don't use the "REMOTE_CONTROL" capability here at Xevo. Hopefully everything works now.

**Comment by fronneburg** [comment]
@LuxoftAKutsan, I believe this PR is complete now. I don't really understand why there is a SEGFAULT in UIGetLanguageResponseTest.Run_LanguageSet_SUCCESS. All tests complete for me in my local environment, so I don't actually know how I would debug this fault.

**Comment by LuxoftAKutsan** [comment]
@fronneburg what flags do you use for building SDL? 
@vkushnirenko-luxoft  could you please provide docker container snapshot for building and unit testing SDL?

**Comment by fronneburg** [comment]
@LuxoftAKutsan We turn ON the following flags for the test build: BUILD_TESTS, BUILD_XEVO_REMOVE_A2DP, REMOTE_CONTROL. All other flags are off. 

It would probably be a great asset to us if we could get a docker container with your environment and tools so we can run our tests interactively and work out problems here before we submit.

**Comment by fronneburg** [comment]
@LuxoftAKutsan I don't understand why the build is failing on Jenkins. It looks like there is a style issue in line 577 of policy_manager_impl_ptu_test.cc, but what's showing up in the output from the style checker isn't actually what's there in that line. Somehow the URL "http://policies.telematics.ford.com/api/policies" has been changed to "http://x.x.x.x:3000/api/1/policies", and the style checker wants that on the previous line.

**Comment by iCollin** [comment]
Closed as this issue was fixed by https://github.com/smartdevicelink/sdl_core/pull/3528

**Pull request closed without merging**